CONTRIBUTING.md

```diff
@@ -55,3 +55,99 @@ Refer to the [README](./README.md) for more details and alternative building uti
 
 1. Make sure unit tests are passing locally with `make test-unit`
 2. Format the source code with `make fmt`
+3. Follow the Commit Message Guidelines
+
+## Commit Message Guidelines
+
+This project has rules for commit messages (based on [Conventional Commits](https://conventionalcommits.org/)). There are several reasons for this:
+
+* Commit messages are more readable, especially when looking through the **project history**.
+* Commit messages describe whether a major, minor or patch change has been introduced (see [semver.org](https://semver.org/))
+* Commit messages can be used to generate a changelog.
+
+### Commit Message Format
```

Comment:
Maybe in this section we could focus on outlining the specifics of the repo's approach to conventional commits (namely the extra tags like `build`, `ci` and referring to the golang packages)

As is this feels more like a small blog post on adopting conventional commits. *hint* *hint*

Reply:
@paolobueno do you mean just the section that describes the reasons why we use conventional commits or?

Reply:
I thought the "type" and "scope" sections were subsections to this one... What I'm suggesting is leave this "Commit Message Format" one out completely, just because it seems to mostly explain what the conventional commits website does. But no harm in having more content.

Reply:
It does repeat some of the content from the conventional commits website but I'd like to explicitly demonstrate it here with appropriate examples too. Main reason: people don't really follow links. I basically took inspiration from the [Angular Commit Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit) but tried to tailor them to our use case.